core/text/table/table.odin

```diff
@@ -251,8 +265,10 @@ write_html_table :: proc(w: io.Writer, tbl: ^Table) {
 }
 
 write_plain_table :: proc(w: io.Writer, tbl: ^Table) {
+	WIDTH_PROC :: unicode_width_proc
+
 	if tbl.dirty {
-		build(tbl)
+		build(tbl, WIDTH_PROC)
```

Comment:
If we are to support custom Width_Proc's they should be stored on the `Table` struct at the very least to avoid this situation. Or just remove custom Width_Proc's. Again, is there a use case?

Reply:
I suppose that if you have a specific font in mind, you could parse its kerning table and glyph info to be more specific. But are you still working in plain text using this package or doing tables in a graphical context?

Reply:
I did consider if the proc should be on the table or not. I concluded that it shouldn't be on the table, because how the table is printed is separate from what data is stored on the table and HTML tables do not measure width.

Reply:
Maybe we should consider `if tbl.dirty` at that point to be an error instead to avoid this. 🤔

Reply:
Do you want `tbl.dirty` done away with?

Reply:
What would you suggest?

Reply:
It's a caching situation. The advantage of having `tbl.dirty` around is that it allows re-printing of the table without needing to recalculate column widths. This is assuming there's a situation where a table has been built and is not often changed, but is shown on more than one or two occasions. However, I think if someone wants to really save time, they're better off writing the table once and storing that as a string and avoiding all of the calculations altogether.

In summary, `dirty` avoids recalculating column widths, but in the end, you're still doing all the work to do the printing, which is the bulk of the work being done (all the calls to `io`'s API which potentially has to branch with each call).

My suggestion is to remove `dirty` and offer string caching as a way to save time, if that's a goal for the programmer.

Reply:
If you don't mind making those changes, then yes I agree that would be a better solution and we should get rid of it.

Reply:
The user could easily do the string caching themselves passing a custom `io.Writer` so I think we can skip that and just remove `dirty`.